### PR TITLE
Feature/dfc sc 138098 align with table design

### PR DESCRIPTION
**Pull request by helaot**
- fix table header

- fix table search bar

- fix table columns

- fix table rows
-  add clear ALL

https://www.figma.com/file/4C6GkYxDROlY3iVJ5keLAq/Demand-Forecast?type=design&node-id=1823-2430&t=fOGZWe1BB21O4ksd-0

### CODE REVIEW

File: commons/_shared/bs-lib-components/ui/src/components/table/BsTableColHeader.vue [hunk 1]
```diff
@@ -1,154 +1,190 @@
...
-                this.lastSearchedValue = newVal[this.col.name];
-            } else {
-                this.lastSearchedValue = "";
-            }
-        }
+        col: Object as PropType<{ label: string; name: string }>,
     },
     methods: {
         sortColumn() {
             if (this.sortable && this.sort) this.sort(this.col);
```
Comment (levkz) [inline commons/_shared/bs-lib-components/ui/src/components/table/BsTableColHeader.vue:126]: - There are three states - unsorted, sorted descending, sorted ascending, you have to account for that in your boolean flag `sortDesc`.
- `sortColIcon` can be a computed property.

Comment (levkz) [reply commons/_shared/bs-lib-components/ui/src/components/table/BsTableColHeader.vue:126]: image

**Review by levkz — CHANGES_REQUESTED** [review]
Good job on the design!

There are a few comments that need to be tended to:
- the sorting logic fails, I've sent you the video in Slack and left a comment.
- virtual scroll is broken, probably some css you've removed, we used parent element height to set the max height of the table, here is quasar link to [virtual scroll](https://quasar.dev/vue-components/table#virtual-scrolling)
- if you test the table on DSS Dataset *(:dss-table-name prop)*  the headers row won't be displayed for some reason.
- There is no way to paginate the DSS Dataset (unless you've implemented infinite scroll).

File: commons/_shared/bs-lib-components/ui/src/components/table/BSTableHeader.vue [hunk 2]
```diff
@@ -8,10 +8,10 @@
                 <BsTableColHeader
                     :sort="props.sort"
                     :col="col"
-                    :searched-cols="searchedCols"
-                    @search-col="(...args) => $emit('search-col', ...args)"
+                    @search-col="activateSearchCol"
                 ></BsTableColHeader>
         </q-th>
+        <q-th auto-width />
```
[the remark this answers is not in the document]
Comment (helaot) [reply commons/_shared/bs-lib-components/ui/src/components/table/BSTableHeader.vue]: - virtual scroll is broken : how can I test it ?
- I wasn't able to test with (:dss-table-name prop) in the project other than the error part as I was getting errors. How did you manage to test it ? 
- There is no way to paginate the DSS Dataset (unless you've implemented infinite scroll). : what do you mean ?

[the remark this answers is not in the document]
Comment (helaot) [reply commons/_shared/bs-lib-components/ui/src/components/table/BSTableHeader.vue]: Update : I managed to test with dss-table-name the virtual scroll didn't seem to be working before my changes

**Review by levkz — APPROVED** [review]
LGTM!

**Pull request merged**